WordPress/Classes/ViewRelated/Blog/BlogDetailsViewController.m

```diff
@@ -514,17 +514,58 @@ - (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInte
 
 #pragma mark - Private methods
 
-- (void)preloadStats
+
+- (void)preloadBlogData
 {
-    NSString *oauthToken = self.blog.authToken;
     WordPressAppDelegate *appDelegate = [WordPressAppDelegate sharedInstance];
     BOOL isOnWifi = [appDelegate.internetReachability isReachableViaWiFi];
     
-    if (isOnWifi && oauthToken) { // only preload on wifi
+    // only preload on wifi
+    if (isOnWifi) {
+        [self preloadStats];
+        [self preloadPosts];
+        [self preloadPages];
+    }
+}
+
+- (void)preloadStats
+{
+    NSString *oauthToken = self.blog.authToken;
+    
+    if (oauthToken) {
         self.statsService = [[WPStatsService alloc] initWithSiteId:self.blog.siteID siteTimeZone:[self.blogService timeZoneForBlog:self.blog] oauth2Token:oauthToken andCacheExpirationInterval:5 * 60];
         [self.statsService retrieveInsightsStatsWithAllTimeStatsCompletionHandler:nil insightsCompletionHandler:nil todaySummaryCompletionHandler:nil latestPostSummaryCompletionHandler:nil commentsAuthorCompletionHandler:nil commentsPostsCompletionHandler:nil tagsCategoriesCompletionHandler:nil followersDotComCompletionHandler:nil followersEmailCompletionHandler:nil publicizeCompletionHandler:nil streakCompletionHandler:nil progressBlock:nil andOverallCompletionHandler:nil];
     }
+}
+
+- (void)preloadPosts
+{
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+    PostService *postService = [[PostService alloc] initWithManagedObjectContext:context];
+    PostListFilterSettings *filterSettings = [[PostListFilterSettings alloc] initWithBlog:self.blog postType:PostServiceTypePost];
```

Comment:
This is really the key reason for creating `PostListFilterSettings`. Now there are no class methods on the view controllers.